[WIP] slots api

kdsolo opened the pull request:
Putting this together made me think of the following: 

  1. What order should the slots files be in? For example, `user/3/slots` has `user/3`, `user/1`, `user/2`, `user/0`...etc. Should `user/0` always just be swapped with the first spot in the array for the next available spot?
  2. Is the slots API necessary right now? Since we do not have a database yet there are only 6 Id's, so I don't see the files changing, and there is no `PUT` to change any of the files either. If we had a `PUT` it would require authentication so users cannot access other users' slots data.

sfoster commented:
The order will be user-determined. I had thought that "self" could always be at slot 0. 
We do need this API soon to explore scenarios in which a user has only assigned or or 2 friends to their device. Or to try out only partially overlapping sets of friends. As long as everyone has the same friends in the same order, its hard to really test out the interactions. Will people be able to remember which slot/led represents which friend? You could go ahead and create users 7,8,9 and mix it up a bit. 

We're not worried about authentication yet because we don't have any real users. All the use cases for this code are for prototyping. 

I'm not sure if we need to create a UI for assigning slots beyond editing the data files in an text/code editor. Of course we'll need it eventually - maybe next week even, but for right now we are focussed exploring the main experience to ensure we are building the right thing

kdsolo commented:
I have changed the files to include different permutations of our current users (I did not add user's 7, 8, and 9). I also updated the `README` to include details on the updated API.

sfoster commented:
Looks great. I agree we don't need a PUT route just yet. This should get us going and able to make the necessary client-side changes.
